### DISCUSSION TITLE
Issues Running mypy in Code App on iPadOS

### DISCUSSION THREAD

**Discussion by verymanlylamp** [thebaselab/codeapp#1023] [Q&A]
Hello everyone,

I'm experiencing issues with running `mypy` for type checking Python files in the Code App on my iPad.

Here's what I've tried so far:
- Running `mypy` with the full path to the file.
- Ensuring that the file name is enclosed in quotes to handle spaces and parentheses.
- Using both relative and absolute paths.
- Attempting to navigate to the file's directory using the `cd` command.

Despite these efforts, `mypy` either can't find the file or can't read it. Here is the command I'm using:

mypy ‘Code/Python Crash Course exercises/Chapter 6 exercises/Rivers (6.4).py’

And here is the error message I receive:

mypy: can’t read file ‘Code/Python Crash Course exercises/Chapter 6 exercises/Rivers (6.4).py’: No such file or directory

I've confirmed that the file exists at the specified location and that the path is correct. The current working directory is `~/Documents`, and the file is within the `Code` directory in that location.

Could this be an issue with how paths are handled in the Code App's terminal on iPadOS, or am I missing something? Any advice or guidance would be greatly appreciated.

Thank you in advance!

**Comment by bummoblizard** [comment]
Hi. I tested mypy on my device and it worked properly. A tip is to use autocomplete in terminal by pressing the tab key on your keyboard. Additionally, try `mypy` with a file located at the `~/Documents` and move further to the subdirectories each step at a time to locate the issue.
